### PR TITLE
Fix/matrix options queries order

### PR DESCRIPTION
**Pull request by Levia**
## Description

Hope this finally fixes the weird ordering issues related to matrix queries and options!

Unfortunately it's not really the kind of fix I wished to apply, but that was the only way (and less painful way?) I found to make it work.
The changes applied are related to sorting the params as they come in from the form, which are being returned in reversed order for some reason; this means that the first option or query created was the one saved the last (so higher ID) resulting in all sorts of issues when adding more queries or options at a later stage (update).
So the items are now also ordered in ascending order (first created first served so to say)

## Notes
[Codebase ticket](https://unep-wcmc.codebasehq.com/projects/ors-maintainance/tickets/12)

### CODE REVIEW

File: app/models/question.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ def self.create_question_answer_type params
   def update_answer_type params
     if params[:part][:answer_type_type] == self.answer_type_type
       #if the answer type is the same, just update the answer_type attributes
-      self.answer_type.update_attributes!(params[:answer_type])
+      self.answer_type.update_attributes!(self.class.sorted_params(params))
```
Comment (lucacug) [inline app/models/question.rb:67]: I wonder if you can just call the method without the `self.class.` bit here

File: app/models/question.rb [hunk 2]
```diff
@@ -75,7 +75,7 @@ def update_answer_type params
       end
       #create thew new answer type based on the params (if they exist. For questions of "same answer type" sections, it doesn't exist.
       if params[:part][:answer_type_type]
-        self.answer_type = params[:part][:answer_type_type].classify.constantize.new(params[:answer_type])
+        self.answer_type = params[:part][:answer_type_type].classify.constantize.new(self.class.sorted_params(params))
```
Comment (lucacug) [inline app/models/question.rb:78]: Same as above

File: app/models/question.rb (hunk 1 above)
Comment (Levia) [reply app/models/question.rb:67]: The method I'm calling is a class method, so I should at least do `class.sorted_params...` as I'm calling it from an instance method instead. I think `self.class` is a bit more readable. What do you think?

Comment (lucacug) [reply app/models/question.rb:67]: yeah makes sense. I think a got confused looking at the previous method, `self.create_question_anser_type`

**Pull request merged**